### ISSUE TITLE
Stopped working

### ISSUE THREAD

**Issue by luizping** [jsanahuja/jquery.instagramFeed#103]
I think IG changed the policy. Now to see a user page, you have to be logged in. So when you run the script, the GET returns the IG Login Page content.

**Comment by mahotilo** [comment]
My IP is now banned, so I can't do a clear check. But yes, something has changed for me too. I will check later if this new IG behaviour is permanent or only for banned IPs.

**Comment by benbracedigital** [comment]
I'm getting the same issue too after a couple of page views Instagram isn't returning any images and in the console I get 

`instagramfeed-1.3.6.min.js?cdv=1056369746:12 Instagram Feed: It looks like YOUR network has been temporary banned because of too many requests. See https://github.com/jsanahuja/jquery.instagramFeed/issues/25`

**Comment by web-assistant** [comment]
I'm also seeing the temporary banned message

**Comment by mahotilo** [comment]
@benbracedigital 
Were you able to see images today before ban? What `cache_time` had you had?

**Comment by benbracedigital** [comment]
Yes I was and I could see the images on a different server this morning but after a couple of refreshes, I get the error message. I've set the `cache_time `to 120.

**Comment by mahotilo** [comment]
It seems like 120min is not enough now. The new [default and recommended](https://github.com/jsanahuja/jquery.instagramFeed/issues/101#issuecomment-758684719) value is 6 hours

**Comment by benbracedigital** [comment]
Sure, happy to increase to 6 hours but I would expect it to be cached for 2 hours previously and still work rather than error and not show images within minutes?

**Comment by mahotilo** [comment]
What version of the script do you have? I can see cached images even in the ban.

**Comment by benbracedigital** [comment]
version 1.3.6

**Comment by mahotilo** [comment]
Caching is only available since 2.0.0

**Comment by benbracedigital** [comment]
Ah, thank you. I'll download the latest version and see what results I get.

**Comment by mahotilo** [comment]
But only after the end of the ban:)

**Comment by benbracedigital** [comment]
I can at least see cached images now. I'm going to deployed to a different (non banned) server to see if that all works.

**Comment by mahotilo** [comment]
Your (browser) IP is banned, not the IP of the server of your page. Personally, I have now been banned on my PC, but I can flawlessly see the script results on my phone with mobile internet.

**Comment by benbracedigital** [comment]
Looks to be working. Thanks for your help

**Comment by mahotilo** [comment]
@luizping 

> I think IG changed the policy. Now to see a user page, you have to be logged in. So when you run the script, the GET returns the IG Login Page content.

I am finally able to check your issue after my last ban ended. The redirect to IG Login Page is only related to the ban. Now, I am able to open and preview any IG page with only offer to register at the bottom.

**Comment by mahotilo** [comment]
Yeah, it looks like these are tough times. IG has banned me right after the first successful page update. And ban will last 2-3 days. Really tough.

**Comment by Federico-Iglesias** [comment]
@mahotilo how are you? Is to possible to avoid this issue by using cache_time setting? Thanks!

**Comment by mahotilo** [comment]
>  Is to possible to avoid this issue by using cache_time setting? Thanks!

The cache_time is the only cure we have, It works. But it is very easy to catch a ban if you simultaneously use the script on different sites and make experiments :)
I am thinking of increasing the cache_time up to 24 h for myself to avoid future bans.

**Comment by pabrul** [comment]
I'm kind of on an "eternal ban", because sometimes the feed appears and sometimes not, the same happens with this error, sometimes it appears on the console and sometimes it doesn't, I really don't know what to do anymore :(

**Comment by pabrul** [comment]
![image](https://user-images.githubusercontent.com/55573363/104962305-ee306100-59b6-11eb-9b64-a8db8da49a65.png)

**Comment by mahotilo** [comment]
@pabrul 

> ...I really don't know what to do anymore :(

First of all, you should update your script to the latest version 2.0.1
This is how your console should look like in case of a ban.

![image](https://user-images.githubusercontent.com/14929385/104965999-8f83db00-59e8-11eb-8152-d7029ac43f97.png)

**Comment by rulloliver** [comment]
Having the same issue, updated to ver 2.0.1 and console reads: 
`jquery.instagramFeed.min.js:98 Instagram Feed: Your network has been temporary banned by Instagram because of too many requests. Consider increasing your 'cache_time'. See https://github.com/jsanahuja/jquery.instagramFeed/issues/25 and https://github.com/jsanahuja/jquery.instagramFeed/issues/101 4`

Increased `cache_time: 360` 4 days ago and no change. 

Any suggestions to proceed?

**Comment by mahotilo** [comment]
> Any suggestions to proceed?

Wait until IG unbans you. Last time, 3 days were enough for me.

**Comment by mahotilo** [comment]
Well, what I have found about the ban from IG.
1. The script and the caching approach work fine. You can update script output on 1-2 pages without being banned.
2. But, @jsanahuja... It seems to me, the demo page that has 4 calls of script in a raw with different username + 1 call with a tag might be the point where the visitors get banned.

**Comment by jsanahuja** [comment]
> 2. But, @jsanahuja... It seems to me, the [demo page](https://www.sowecms.com/demos/jquery.instagramFeed/index.html) that has 4 calls of script in a raw with different username + 1 call with a tag might be the point where the visitors get banned.

Maybe we should change it to use the same username on all the examples. The tag call doesn't matter because it never gets banned as far as I know

**Comment by mahotilo** [comment]
> The tag call doesn't matter because it never gets banned as far as I know

I also think so.

> Maybe we should change it to use the same username on all the examples.

Or maybe the function should be called only when the corresponding tab is opened? Barcelona should be kept:)

**Comment by jsanahuja** [comment]
> Or maybe the function should be called only when the corresponding tab is opened? 
Mmm, way better

> Barcelona should be kept:)
Hahaha, for sure 😄

**Comment by Remling97** [comment]
Just have a question for clarification, if i get banned on one page, will i be banned on all pages that use this script, or is the ban on a per page basis? or is it on a per instagram page basis?

**Comment by mahotilo** [comment]
I believe, it is a ban on your IP.

**Comment by mahotilo** [comment]
One last time (hopefully)
My clear experiment
1. No ban, more than 6 h after the last update
2. My first test page (1 "name" script instance)
3. Demo page (4 "name" script instances)
4. My second test page (1 "name" script instance)
5. Ban

Magic number is 6.

I think this issue should be closed. It has nothing to do with script and caching. It is about the number of script calls.

**Comment by m0nclous** [comment]
#104 Feature/proxy this may solve the problem.
But it needs testing. This is still a draft.
I can post a backend heroku, everything is easy there.

**Comment by mahotilo** [comment]
Hm, it doesn't work for me
https://instagram-feed-proxy.herokuapp.com/?url=https://www.instagram.com/instagram

![image](https://user-images.githubusercontent.com/14929385/105726949-3f9a9c00-5f33-11eb-865b-d8978329dae5.png)

**Comment by m0nclous** [comment]
> Hm, it doesn't work for me
> https://instagram-feed-proxy.herokuapp.com/?url=https://www.instagram.com/instagram
> 
> ![image](https://user-images.githubusercontent.com/14929385/105726949-3f9a9c00-5f33-11eb-865b-d8978329dae5.png)

This is normal. You only need to get _sharedData
You only need to get _sharedData from html response

**Comment by m0nclous** [comment]
@mahotilo I hope you understand what I'm talking about. I'm not very good at translating into English, sorry.

**Comment by mahotilo** [comment]
@m0nclous

> I hope you understand what I'm talking about.

Not a problem.

> This is normal. You only need to get _sharedData from html response.

Yes, I can reproduce your example with _sharedData. Thx.

**Comment by mahotilo** [comment]
@m0nclous

Well, I have just tested by changing the host attribute to proxy.
`                'host': 'https://instagram-feed-proxy.herokuapp.com/?url=https://www.instagram.com/',`

It works well. Hurrah!
But why do we need to introduce two new attributes just to change the value of the `host`?
Wouldn't it be easier to add the proxy address to the documentation as a recommended alternative to the host value?

P.S.
Funny. Today I had to download over 20 IG pages to get banned :)

**Comment by mahotilo** [comment]
As I see it, this decision has been described here https://lopodofoodservice.com/assets/js/plugins/instagramfeed/

**Comment by m0nclous** [comment]
> But why do we need to introduce two new attributes just to change the value of the `host`?

Yeah, this is the best solution

**Comment by m0nclous** [comment]
> Funny. Today I had to open over 20 IG pages to get banned :)

The whole problem is session related.

You can't create more than two (or three, haven't tested it) sessions from one IP. If you clear your cookies, you will lose the session ID, thereby losing access to the instagram view, for example through incognito mode.

But by saving your session, you can view without restrictions. On heroku there is no request proxying through third-party servers, there is just a request with the same session ID in the cookie.

**Comment by mahotilo** [comment]
> But by saving your session, you can view without restrictions. 

Hm, I guess this is the number of username views per day that is limited by IG for one IP. Today to get a ban I clicked on near 20 profiles on https://www.instagram.com/explore/tags/paradise. It was in incognito mode in one session.

**Comment by m0nclous** [comment]
> Hm, I guess this is the number of username views per day that is limited by IG for one IP. Today to get a ban I clicked on near 20 profiles on https://www.instagram.com/explore/tags/paradise. It was in incognito mode in one session.

My mistake. It turns out I was using my account session. 🤦‍♂️
Heroku sends requests from my session and returns the result.

How many views will last before blocking or captcha?
I think I need to install and test the recorder. But at least this solves the problem, especially if you are using caching.

**Comment by mahotilo** [comment]
> How many views will last before blocking or captcha?

So far, I have run the script with the proxy in the host over 20 times in different incognito sessions. It still works like a charm!

**Comment by m0nclous** [comment]
The result of the discussion: all you need to do is specify `'host': 'https://instagram-feed-proxy.herokuapp.com/?url=https://www.instagram.com/'` (or another proxy of your choice) in instagramFeed settings.

I will continue to explore caching, proxying and process optimization on heroku app, after which I will post it to the public.

**Comment by mahotilo** [comment]
@m0nclous, @jsanahuja 
Would you make a PR for the documentation?
Maybe here?
![image](https://user-images.githubusercontent.com/14929385/105755566-22c29080-5f54-11eb-8b93-0ffe3882013a.png)

**Comment by m0nclous** [comment]
@mahotilo I figured out how to do without a proxy.

To do this, you need to know the user id, from where the pictures will be taken.
`https://www.instagram.com/web/search/topsearch/?query=<username>`

Then we make a request with the found id
`https://instagram.com/graphql/query/?query_id=17888483320059182&variables={"id":"<user_id>","first":20,"after":null}`

Thus, you can keep the ability to insert instagram via username.

The ID can be cached in order not to receive it all the time.

I don't see any blocking from instagram.
I should be able to implement this in 1-2 days, including the vanilla plugin.

The question remains how to get data from the tag without blocking.

https://stackoverflow.com/questions/17373886/how-can-i-get-a-users-media-from-instagram-without-authenticating-as-a-user/47243409#47243409
https://stackoverflow.com/questions/11796349/instagram-how-to-get-my-user-id-from-username/52808289#52808289
https://github.com/jsanahuja/InstagramFeed/issues/49#issuecomment-758598459

**Comment by mahotilo** [comment]
It seems that the tag is never blocked.

**Comment by m0nclous** [comment]
Then there is no problem. I will try to find time tomorrow and implement what I wrote above for testing.
It looks like the issue will be resolved.

**Comment by web-burn** [comment]
@mahotilo, any news about fix?  
Currently the fix with proxy host: "https://instagram-feed-proxy.herokuapp.com/?url=https://www.instagram.com/'" doesn't work.

**Comment by mahotilo** [comment]
> Currently the fix with proxy host: "https://instagram-feed-proxy.herokuapp.com/?url=https://www.instagram.com/'" doesn't work.

Unfortunately, that is to be expected. IG is hard to cheat.
But caching still does the trick. It is not perfect but quite acceptable.

**Comment by web-burn** [comment]
> > Currently the fix with proxy host: "https://instagram-feed-proxy.herokuapp.com/?url=https://www.instagram.com/'" doesn't work.
> 
> Unfortunately, that is to be expected. IG is hard to cheat.
> But caching still does the trick. It is not perfect but quite acceptable.
> 
> @m0nclous, do you have any new findings?

If i will use a value "1440" for the "cache_time" then single IP will not be banned?

**Comment by mahotilo** [comment]
As I understand it.
IG wants you to subscribe and log in. So, it limits the number of interactions of an unregistered user with images and profiles (names). It doesn't matter if it is several images in one profile or just one image in several profiles.
The script acts as unregistered user, so your IP is the subject of ban. It is not the IP of page server.

Thus, if you use a big cache time, you protect yourself from being banned when interacting with the same profile in the same page. But, if you run multiple scrips with different profiles (username) within a short period of time, you will still be banned, no mater how big the cache time is.

**Comment by m0nclous** [comment]
> @mahotilo, any news about fix?
> Currently the fix with proxy host: "https://instagram-feed-proxy.herokuapp.com/?url=https://www.instagram.com/'" doesn't work.

![image](https://user-images.githubusercontent.com/18490365/106376098-114a1000-63bc-11eb-9182-0560a2a58172.png)
Yeah, they still blocked the session after a while

**Comment by m0nclous** [comment]
> @m0nclous, do you have any new findings?

I implemented (locally) what I said above, but I got the same problem - Instagram blocks requests to
> `https://www.instagram.com/web/search/topsearch/?query=<username>`

after about 6 requests

BUT! Instagram doesn't block it
> `https://instagram.com/graphql/query/?query_id=17888483320059182&variables={"id":"<user_id>","first":20,"after":null}`

Thus, we can avoid errors if we set the target profile using ID, but to completely solve the issue, we need to find a way to obtain ID based on username without blocking.

**Comment by m0nclous** [comment]
We can receive data through `https://www.instagram.com/<username>/?__a=1` but this path is blocked after about 4 requests

**Comment by mahotilo** [comment]
> but this path is blocked after about 4 requests

At least we can create an instruction on how to get this ID manually.

**Comment by m0nclous** [comment]
@mahotilo 
`https://instagram.com/graphql/query/?query_id=17888483320059182&variables={"id":"<user_id>","first":20,"after":null}`
Doesn't return profile info, such as profile_pic_url/username/biography

What to do with it?

**Comment by mahotilo** [comment]
Hmm. Very intriguing. While I was looking for an answer to your question I found this article

https://medium.com/@carloshenriquereis_17318/how-to-get-data-from-a-public-instagram-profile-edc6704c9b45

1. author describes the same problem and the same approach to solving
2. he gets missed user data from request https://www.instagram.com/<username>/?__a=1
3. he gets more than 12 images with offset in request

His demo is at the end of 
https://carlos-henreis.github.io/

**Comment by m0nclous** [comment]
@mahotilo Yes, I read that too. The request is blocking just like the others
We can cache this information for a long time, but this is not a solution to the problem, since caching is not a solution for many users, if you look at this issue

**Comment by mahotilo** [comment]
I believe that hiding data of user account is a part of IG's policy. So there is a little chance of finding a long-lasting trick.

But for many users, this script is just a handy way to get an easy-to-update picture gallery on the site. I have even hid a user info in some projects. 
So this might be an option:
- use cached full data by username with delay and possible pause for ban
- use instantly updated images by ID without username info. (If you can find the ID in JSON, you can insert the profile image and name yourself)

Personally, I think caching is a great solution. The only real alternative is to use IG API.

**Comment by web-burn** [comment]
Now i noticed that Instagram bans me always. I changed my IP to another IP but new IP is banned already.

**Comment by web-burn** [comment]
Hello,

The plugin stopped work.
<img width="1249" alt="Screenshot_1" src="https://user-images.githubusercontent.com/22581200/108832710-55a88480-75e5-11eb-9ac1-47dcf6344592.png">

**Comment by jsanahuja** [comment]
It looks like they have added again a CORS restriction but only in some of their servers. We will test and implement the graphql solution soon

**Comment by web-burn** [comment]
@jsanahuja many thanks! Can you say as soon new solution will be ready? (approximate time of course). Sorry for the indiscreet question.

**Comment by jsanahuja** [comment]
It is ready @web-burn.

Unfortunately GraphQL API is also affected by the arbitrary CORS issues. I have also added the `max_tries` parameter to retry when that happens.

**Comment by web-burn** [comment]
@jsanahuja It is should updated plugin version to 2.1.1 on my site? Is this fix the CORS issue?

**Comment by jsanahuja** [comment]
> @jsanahuja It is should updated plugin version to 2.1.1 on my site? Is this fix the CORS issue? 

CORS issue is not something we can fix client-side. As it is arbitrary (only some of their cluster servers have a CORS restriction) we have implemented a policy of retries.
You should definitely update to the 2.1.1 but the real solution would be using a proxy and the graphQL (`user_id` parameter)

**Comment by colinchadwick** [comment]
It isn’t working anymore.

**Comment by romanklabal** [comment]
So since the idiots at IG introduced CORS, there is simply no way to reliably load the last 12 IG posts based on the username any longer?

The only solution is to use graphQL with `user_id`, which we.. can not get programatically and have to use manual workarounds to get it first?

**Comment by colinchadwick** [comment]
I don’t get why they don’t allow to access data that’s already public, and their shitty Basic Display API doesn’t even allow me to get the url of a profile picture. That’s ridiculous.

**Comment by mahotilo** [comment]
They want to know personally everyone who opens their site :)

**Comment by colinchadwick** [comment]
Actually, I've just found a (very tricky, but working) way to bypass the IG CORS policy – use this as the host:
`'host': "https://images" + ~~(Math.random() * 3333) + "-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=https://www.instagram.com/",`

**Comment by piotrmerton** [comment]
> Actually, I've just found a (very tricky, but working) way to bypass the IG CORS policy – use this as the host:
> `'host': "https://images" + ~~(Math.random() * 3333) + "-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=https://www.instagram.com/"`

Thanks!

**Comment by iennetastic** [comment]
> Actually, I've just found a (very tricky, but working) way to bypass the IG CORS policy – use this as the host:
> `'host': "https://images" + ~~(Math.random() * 3333) + "-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=https://www.instagram.com/"`

This works for me too! Thank you! <3

**Comment by mahotilo** [comment]
> 'host': "https://images" + ~~(Math.random() * 3333) + "-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=https://www.instagram.com/"

Indeed, magic works. But, if I try to open the generated URL in the browser, I get the message ```We're sorry... ... but your computer or network may be sending automated queries. To protect our users, we can't process your request right now.```

So the question is, should I just not open this URL directly in the browser, or will we get banned by Google this time?

**Comment by colinchadwick** [comment]
I would say it’s unlikely to get banned by Google, I‘ve seen other use this at a fairly high scale. I just hope that this doesn’t get banned by IG.

**Comment by introzen** [comment]
> Actually, I've just found a (very tricky, but working) way to bypass the IG CORS policy – use this as the host:
> `'host': "https://images" + ~~(Math.random() * 3333) + "-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=https://www.instagram.com/"`

Works like a charm! Cheers!

**Comment by m0nclous** [comment]
@mahotilo
> But, if I try to open the generated URL (host + username) in the browser, I get the message `We're sorry... ... but your computer or network may be sending automated queries. To protect our users, we can't process your request right now.`

Google checks the referer header

![image](https://user-images.githubusercontent.com/18490365/109644359-95401500-7b7f-11eb-8f89-d50558a55788.png)

![image](https://user-images.githubusercontent.com/18490365/109644567-d59f9300-7b7f-11eb-8e42-9bdff34219d4.png)

**Comment by mahotilo** [comment]
@m0nclous 
So as of now, this is kind of a _permanent_ solution! Can we call it the _Google proxy_? :)

**Comment by KessellJames** [comment]
The proxy works thanks but I’m finding extra margin now between image rows - can I fix this?

**Comment by introzen** [comment]
No longer appears to be working.

![image](https://user-images.githubusercontent.com/25616937/109652762-79794680-7b60-11eb-8a16-9065ee44997b.png)

**Comment by m0nclous** [comment]
> > Actually, I've just found a (very tricky, but working) way to bypass the IG CORS policy – use this as the host:
> > `'host': "https://images" + ~~(Math.random() * 3333) + "-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=https://www.instagram.com/"`
> 
> No longer appears to be working.
> 
> ![image](https://user-images.githubusercontent.com/25616937/109652762-79794680-7b60-11eb-8a16-9065ee44997b.png)

@introzen This works if you make an ajax request or add a referer header manually https://github.com/jsanahuja/jquery.instagramFeed/issues/103#issuecomment-788854136

**Comment by Creative-developers** [comment]
https://images" + ~~(Math.random() * 3333) + "-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=

This is not working for following link
https://www.instagram.com/graphql/query/?query_hash=" + queryHash + "&variables=" + variables;

**Comment by jsanahuja** [comment]
Upgrade to [3.0.0](https://github.com/jsanahuja/jquery.instagramFeed/releases/tag/3.0.0) and use this https://github.com/jsanahuja/jquery.instagramFeed/issues/103#issuecomment-788260968

----

> https://images" + ~~(Math.random() * 3333) + "-focus-opensocial.googleusercontent.com/gadgets/proxy?container=none&url=
> 
> This is not working for following link
> https://www.instagram.com/graphql/query/?query_hash=" + queryHash + "&variables=" + variables;

Right. We are not able to use GraphQL anymore unless we use a real proxy.

**Comment by Mikhail-MM** [comment]
@jsanahuja @introzen 

This looks almost like we are spoofing/tricking IG by manufacturing the header - I would not be surprised if the solution goes down again, no?

**Comment by web-burn** [comment]
@jsanahuja i noticed this error now. The instagram feed doesn't displayed. Browser firefox. I use the latest version 3.0.0. 
 <img width="945" alt="Screenshot_1" src="https://user-images.githubusercontent.com/22581200/109949162-7ca53b80-7cf4-11eb-9697-c2674acda88b.png">

**Comment by mahotilo** [comment]
@web-burn 
Is not it related to https://github.com/jsanahuja/jquery.instagramFeed/issues/117?

**Comment by web-burn** [comment]
@mahotilo  sometimes i see this error on real hosting too. 
Also i noticed the next error when i change username to another in plugin settings:
<img width="1260" alt="Screenshot_1" src="https://user-images.githubusercontent.com/22581200/109952370-318d2780-7cf8-11eb-8ccf-abfe6ced23d2.png">
<img width="672" alt="Screenshot_2" src="https://user-images.githubusercontent.com/22581200/109952373-3225be00-7cf8-11eb-8887-472bd30cd35d.png">

**Comment by web-burn** [comment]
@mahotilo  Sometimes i see the error on real hosting too. 
Also i noticed the next error when i change username in plugin settings:
<img width="1255" alt="Screenshot_1" src="https://user-images.githubusercontent.com/22581200/109953346-5fbf3700-7cf9-11eb-9986-4b937137c271.png">
<img width="735" alt="Screenshot_2" src="https://user-images.githubusercontent.com/22581200/109953354-6188fa80-7cf9-11eb-8f1d-a665c9d17205.png">

**Comment by mahotilo** [comment]
> Sometimes i see the error on real hosting too.

I do not have it here. Could you describe a condition to reproduce?

> Also i noticed the next error when i change username in plugin settings:

It can be if `parse_caption` returns `false`. It should not be an isolated error, just a consequence of an early problem.

**Comment by web-burn** [comment]
@mahotilo I noticed this error only for some instagram accounts. Please try the next user name:
modern_presta

**Comment by m0nclous** [comment]
> @jsanahuja i noticed this error now. The instagram feed doesn't displayed. Browser firefox. I use the latest version 3.0.0.
> <img alt="Screenshot_1" width="945" src="https://user-images.githubusercontent.com/22581200/109949162-7ca53b80-7cf4-11eb-9697-c2674acda88b.png">

This happens when Google responds with an error, because in this case it does not send the header "Access-Control-Allow-Origin"
To understand the reason - you need to reproduce the problem and see the answer from Google manually.

**I think this solution is still crude and more logging needs to be added.**

![image](https://user-images.githubusercontent.com/18490365/109961871-b71ed080-7d14-11eb-8f77-1d82b24dabd0.png)

![image](https://user-images.githubusercontent.com/18490365/109961737-88085f00-7d14-11eb-96e7-564c1af3e5b4.png)

**Comment by mahotilo** [comment]
> @mahotilo I noticed this error only for some instagram accounts. Please try the next user name:
> modern_presta
> 
> Update: The error occur when instagram account doesn't have bio.

I have checked, there is the same error here.
But, it has happened to me before too. It is related to the first VIDEO in the row.

**Comment by m0nclous** [comment]
> > @mahotilo I noticed this error only for some instagram accounts. Please try the next user name:
> > modern_presta
> > Update: The error occur when instagram account doesn't have bio.
> 
> I have checked, there is the same error here.
> But, it has happened to me before too. It is related to the first VIDEO in the row.

This happens when parse_caption() dont finds a caption and returns false in escape_string where the string is expected.
I will fix it.

![image](https://user-images.githubusercontent.com/18490365/109965419-24ccfb80-7d19-11eb-8d26-e8c6b6e9aeea.png)

![image](https://user-images.githubusercontent.com/18490365/109965350-0cf57780-7d19-11eb-9632-b070c5e05060.png)

**Comment by mahotilo** [comment]
We definitely should have created a separate new issue for this :)

**Comment by m0nclous** [comment]
> We should definitely have created a separate new issue for this :)

Can you do it please?

**Comment by mahotilo** [comment]
Please, let us continue to discuss this error here https://github.com/jsanahuja/jquery.instagramFeed/issues/120

**Comment by jsanahuja** [comment]
> Access-Control-Allow-Origin

This is because you are running the script without a web server. The solution is setting the 'Referer' header but web browsers won't allow us to do that with Javascript for security reasons

**Comment by thomashdk** [comment]
ALso gettings this. Using 3.0.0 whit the "fix" host 

![image](https://user-images.githubusercontent.com/6445636/109970200-0fde7100-7cf5-11eb-9f4b-741bd2f03445.png)

**Comment by mahotilo** [comment]
@thomashdk, I can only quote for you

> this is because you are running the script without a web server

**Comment by thomashdk** [comment]
On my site I get this ?

![image](https://user-images.githubusercontent.com/6445636/109975021-75812c00-7cfa-11eb-8386-39e768e6ffff.png)

**Comment by mahotilo** [comment]
@thomashdk, can you see the script output for the demo page https://www.sowecms.com/demos/jquery.instagramFeed/index.html?

**Comment by thomashdk** [comment]
![image](https://user-images.githubusercontent.com/6445636/109975794-4fa85700-7cfb-11eb-9726-7274d306a4e7.png)
 hmm no ? @mahotilo 

> @thomashdk, can you see the script output for the demo page https://www.sowecms.com/demos/jquery.instagramFeed/index.html?

**Comment by mahotilo** [comment]
@thomashdk 
Hmm, it works for me here

**Comment by thomashdk** [comment]
@mahotilo i'm getting the images but also all the errors.. 

Weird..

**Comment by mahotilo** [comment]
@thomashdk 
> Weird..

Not exactly. Some of these errors are caused by fair try to load data directly from IG in case they will change their minds.
Other errors are related to too big amount of images in the TAG example (see https://github.com/jsanahuja/jquery.instagramFeed/issues/119)

**Comment by mahotilo** [comment]
@thomashdk, to get rid of such errors in the console, follow this recommendation.
![image](https://user-images.githubusercontent.com/14929385/109982041-2cd16f00-7d0a-11eb-9c65-c077d104fcfe.png)

**Comment by thomashdk** [comment]
> @thomashdk, to get rid of such errors in the console, follow this recommendation.
> ![image](https://user-images.githubusercontent.com/14929385/109982041-2cd16f00-7d0a-11eb-9c65-c077d104fcfe.png)

Hi 

hmm have already tried with that..

**Comment by mahotilo** [comment]
Are you using the script with TAG?

**Comment by thomashdk** [comment]
i'm using Username

**Comment by thomashdk** [comment]
@mahotilo If I change the username to "Instragram" I get their feed.

If I changed it to my own "thomas___dk" then it dont..

**Comment by thomashdk** [comment]
You don't have the hostname specified?

Arh you using this ? 
https://cdnjs.cloudflare.com/ajax/libs/jquery.instagramFeed/3.0.0/jquery.instagramFeed.min.js

**Comment by mahotilo** [comment]
> You don't have the hostname specified?

Yep. But it is not a problem.

> Arh you using this ?

I am using  version 3.0.3.

**Comment by web-burn** [comment]
Hello,

I noticed that post images URLs are unavailable very often.
<img width="1292" alt="Screenshot_1" src="https://user-images.githubusercontent.com/22581200/113484476-8d92c980-94b9-11eb-8e5e-9da7bf9917b5.png">
![Screenshot_2](https://user-images.githubusercontent.com/22581200/113484479-8ec3f680-94b9-11eb-808b-e21278593f67.png)

**Comment by unlockdesgn** [comment]
Suddenly instagram feed down ! 
Example:
https://www.jqueryscript.net/demo/Instagram-Photos-Without-API-instagramFeed/

**Comment by tiktiddies** [comment]
> Suddenly instagram feed down !
> Example:
> https://www.jqueryscript.net/demo/Instagram-Photos-Without-API-instagramFeed/

same error in demo and my website https://www.tiktiddies.com/ it's not a problem of IP banned but a problem of "Access to XMLHttpRequest at 'https://www.instagram.com/XXXXXXX/' from origin 'https://www.tiktiddies.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource."

**Comment by unlockdesgn** [comment]
What is the solution ?
